x-pack/plugins/monitoring/server/lib/cluster/get_index_patterns.test.ts

```diff
@@ -14,89 +15,182 @@ const getConfigWithCcs = (ccsEnabled: boolean) => {
       ccs: {
         enabled: ccsEnabled,
       },
+      logs: {
+        index: 'filebeat-*',
+      },
     },
   } as MonitoringConfig;
 };
 
+type TestTuple = [DS_INDEX_PATTERN_TYPES | undefined, string];
 describe('getNewIndexPatterns', () => {
   beforeEach(() => {
     jest.resetModules();
   });
-  it('returns local elasticsearch index patterns when ccs is enabled (default true) and no ccs payload', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(true),
-      moduleType: 'elasticsearch',
-    });
-    expect(indexPatterns).toBe('.monitoring-es-*,metrics-elasticsearch.*-*');
-  });
-  it('returns ecs only elasticsearch index patterns when specifying ecsLegacyOnly: true', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(true),
-      moduleType: 'elasticsearch',
-      ecsLegacyOnly: true,
-    });
-    expect(indexPatterns).toBe('.monitoring-es-8-*,metrics-elasticsearch.*-*');
-  });
-  it('returns local kibana index patterns when ccs is enabled with no ccs payload', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(true),
-      moduleType: 'kibana',
-    });
-    expect(indexPatterns).toBe('.monitoring-kibana-*,metrics-kibana.*-*');
-  });
-  it('returns logstash index patterns when ccs is enabled and no ccs payload', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(true),
-      moduleType: 'logstash',
-    });
-    expect(indexPatterns).toBe('.monitoring-logstash-*,metrics-logstash.*-*');
-  });
-  it('returns beats index patterns when ccs is enabled and no ccs payload', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(true),
-      moduleType: 'beats',
-    });
-    expect(indexPatterns).toBe('.monitoring-beats-*,metrics-beats.*-*');
-  });
-  it('returns elasticsearch index patterns with dataset', () => {
+  it.each<TestTuple>([
+    [undefined, '.monitoring-es-*,metrics-elasticsearch.*-*'],
+    ['metrics', '.monitoring-es-*,metrics-elasticsearch.*-*'],
+    ['logs', 'filebeat-*,logs-elasticsearch.*-*'],
+  ])(
+    'returns local elasticsearch index patterns  when ccs is enabled (default true) and no ccs payload and type %s',
+    (type, expected) => {
+      const indexPatterns = getNewIndexPatterns({
+        type,
+        config: getConfigWithCcs(true),
+        moduleType: 'elasticsearch',
```

Comment:
I kind of wanted to put `moduleType` in the TestTuple, but I can see the `ecsLegacyOnly: true`. That got me wondering what `ecsLegacyOnly: true/false` is even really _for_, but that's beyond the scope of this PR I think.